Bug fix ets get prediction

pratyushsharan opened the pull request:
- [x ] closes #7018 
- [x ] tests added / passed. 
- [x ] code/documentation is well formatted.  
- [x ] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>
Fixed the issue where the ETSModel get_prediction method fails when start is greater than the last index. Modified test case to reflect this use case, which would fail on master and would pass once this branch is merged.

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

bashtage commented:
Could you parameterize the test using different start values including one in-sample, one at the end of the sample, one just out of sample, and one well out of sample?

Thanks,
Kevin

pratyushsharan commented:
> Could you parameterize the test using different start values including one in-sample, one at the end of the sample, one just out of sample, and one well out of sample?

Added the suggested tests. Have also added a test case where `start` == `end`, as a sanity check.

bashtage commented:
Many thanks.
